File: LeetCode/1827. 最少操作使数组递增.py
class Solution:
    def minOperations(self, nums: list[int]) -> int:
        cnt = 0
        if len(nums) == 0 or len(nums) == 1:
            return cnt

        for i in range(len(nums) - 1):
            if nums[i + 1] <= nums[i]:
                tmp = nums[i] - nums[i + 1]
                cnt += (tmp + 1)
                nums[i + 1] += (tmp + 1)
        return cnt

File: LeetCode/1827. 最少操作使数组递增.cpp
class Solution
{
public:
    static int minOperations(vector<int> &nums)
    {
        int tmp = 0, cnt = 0;
        for (int i = 0; i < nums.size() - 1; ++i)
        {
            if (nums[i] >= nums[i + 1])
            {
                tmp = nums[i] - nums[i + 1] + 1;
                nums[i + 1] += tmp;
                cnt += tmp;
            }
        }
        return cnt;
    }
};